solver/llbsolver/solver.go

```diff
@@ -233,11 +233,12 @@ func (s *Solver) recordBuildHistory(ctx context.Context, id string, req frontend
 			span, ctx := tracing.StartSpan(ctx, fmt.Sprintf("create %s history provenance", name))
 			defer span.End()
 
-			prc, err := NewProvenanceCreator(ctx2, cap, res, attrs, j, usage)
+			// TODO: use provenance slsa v1 for build history?
+			pc, err := NewProvenanceCreator(ctx2, provenancetypes.ProvenanceSLSA02, cap, res, attrs, j, usage)
```

Comment:
For backward compatiblity with tools using v0.2.

Reply:
@tonistiigi As discussed offline we would want an opt-in to enable SLSA v1 for provenance in build history. I guess this would not be through the new `version` attribute but a builtin-var like `BUILDKIT_HISTORY_PROVENANCE_SLSA`?

Reply:
Added opt-in for provenance slsa v1 for build record history